tests/test_items.py

```diff
@@ -0,0 +1,190 @@
+
+
+def test_product_serialization():
+    product = Product(**_PRODUCT_ALL_KWARGS)
+    adapter = ItemAdapter(product)
+    serialized_product = adapter.asdict()
+    assert serialized_product == {
+        "additionalProperties": [dict(name="foo", value="bar")],
+        "aggregateRating": dict(
+            bestRating=5.0,
+            ratingValue=2.5,
+            reviewCount=123,
+        ),
+        "availability": "InStock",
+        "brand": dict(name="Ka-pow"),
+        "breadcrumbs": [
+            dict(name="Level 1", link="http://example.com/level1"),
+            dict(name="Level 2", link="http://example.com/level1/level2"),
+        ],
+        "canonicalUrl": "https://example.com/product22",
+        "color": "white",
+        "currency": "USD",
+        "currencyRaw": "$",
+        "description": "Full freshness all over the fridge\n5 Conversion Modes on demand\nSuper Cooling Plus™",
+        "descriptionHtml": (
+            "<article><p>Full freshness all over the fridge</p>"
+            "<p>5 Conversion Modes on demand</p>"
+            "<p>Super Cooling Plus&trade;</p></article>"
+        ),
+        "features": ["Easily store fragile products.", "Bluetooth connectivity."],
+        "gtin": [dict(type="foo", value="bar")],
+        "images": [
+            dict(
+                dataUrl=(
+                    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFBAMAAAB"
+                    "/QTvWAAAAFVBMVEXMzMyWlpaxsbG3t7e+vr7FxcWjo6PZoQRwAAAACXBIWXMA"
+                    "AA7EAAAOxAGVKw4bAAAAEklEQVQImWNgQALKDgwMYQwMAAOLALoJbp2PAAAAA"
+                    "ElFTkSuQmCC"
+                ),
+                url=None,
+            ),
+            dict(url="http://example.com/image1.png", dataUrl=None),
+        ],
+        "mainImage": dict(url="http://example.com/image1.png", dataUrl=None),
+        "metadata": dict(
+            dateDownloaded="20221231T130154Z",
+            probability=1.0,
+        ),
+        "mpn": "HSC0424PP",
+        "name": "White two-door refrigerator",
+        "price": "9999.99",
+        "productId": "A123DK9823",
+        "regularPrice": "11999.99",
+        "size": "XL",
+        "sku": "A123DK9823",
+        "style": "polka dots",
+        "url": "https://example.com/?product=product22",
+        "variants": [
+            {
+                "additionalProperties": None,
+                "availability": None,
+                "canonicalUrl": None,
+                "color": None,
+                "currency": None,
+                "currencyRaw": None,
+                "gtin": None,
+                "images": None,
+                "mainImage": None,
+                "mpn": None,
+                "name": None,
+                "price": None,
+                "productId": None,
+                "regularPrice": None,
+                "size": None,
+                "sku": None,
+                "style": None,
+                "url": None,
+            },
```

Comment:
Is this OK for an empty variant? Or should we make it so that unset fields are excluded from serialization somehow?

If we do _not_ exclude fields, should we have a value other than `None` as default for some fields (e.g. `[]` for lists)?

Reply:
We should exclude the following from export:

* None values
* empty lists
* empty dictionaries